workspaces/julieanna/ims_latlong.ipynb

```diff
@@ -0,0 +1,729 @@
+    "    colors = [\"#E500E5\", \"#0066FF\", \"#01FF00\", \"#FFC100\", \"#E50000\"]\n",
+    "    cmap = ListedColormap(colors, name=\"custom_colormap\", N=len(colors))\n",
+    "\n",
+    "    # Plotting the sea ice extent with the custom colormap\n",
+    "    plt.figure(figsize=(10, 10))\n",
+    "    plt.imshow(sie, cmap=cmap, vmin=0, vmax=len(colors) - 1)\n",
+    "    plt.title(\"Sea Ice Extent around the Arctic Circle\", fontsize=12)\n",
+    "    plt.axis(\"off\")\n",
+    "\n",
+    "    # Adding a colorbar with labels for different surface types\n",
+    "    cbar = plt.colorbar(ticks=range(len(colors)))\n",
+    "    cbar.ax.set_yticklabels(\n",
+    "        [\n",
+    "            \"Outside Northern Hemisphere\",\n",
+    "            \"Open Water\",\n",
+    "            \"Land without Snow\",\n",
+    "            \"Sea or Lake Ice\",\n",
+    "            \"Snow Covered Land\",\n",
+    "        ]\n",
+    "    )\n",
+    "    cbar.set_label(\"Surface Type\")\n",
+    "    if marker:\n",
+    "        plt.scatter(marker[0], marker[1], s=500, c=\"green\", marker=\"o\")\n",
+    "\n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "e72d5969-6057-40e1-9b5f-c906ba218b6b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def find_closest_index_in_grid(target: float, coordinates: np.array) -> int:\n",
+    "    \"\"\"\n",
+    "    Given a target coordinate in projected coordinates (x or y) and the list of coordinates, return the index of the closest number in the coordinates.\n",
+    "    Assumes a 1km grid.\n",
+    "    \"\"\"\n",
+    "    assert np.allclose(coordinates % 1000, 500)\n",
+    "    start = coordinates[\n",
+    "        0\n",
+    "    ]  # Assume that the first element of coordinates is the minimum number in the list\n",
+    "    # -12287500.0 for IMS 1km data.\n",
+    "\n",
+    "    # Define the step size\n",
+    "    grid_resolution = 1000  # meters\n",
+    "\n",
+    "    # Calculate the index of the closest number\n",
+    "    index = int((target - start) // grid_resolution)\n",
+    "\n",
+    "    # If the solution is correct, the target and the found value should never be more than the grid_resolution apart.\n",
+    "    assert abs(coordinates[index] - target) < grid_resolution\n",
+    "    return index"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ccb3a114-8398-4ad1-895c-d6215c46427e",
+   "metadata": {},
+   "source": [
+    "## Load file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "88af5147-c23e-4f1a-a734-343594f34921",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "file_path = \"../../data/ims_netcdf_1km_cropped_4_000_000m_window/2015/ims2015164_1km_v1.3_grid4000.nc\"\n",
+    "file_path = \"../../data/ims_netcdf_1km_cropped_2_000km_window/2015/ims2015164_1km_v1.3_grid2000_74,0lat_-140,0lon.nc\"\n",
+    "file_path = \"../../data/ims_1km/2015/ims2015164_1km_v1.3.nc\""
```

Comment:
which data is used for the cells that was run below?

Reply:
Hopefully it's clearer now! I added some more explanations and hopefully every cell is easy to follow now.

Reply:
Fixed in https://github.com/NOAA-HES-Capstone/icedyno/pull/53/commits/0007533e46a05fd21bde9b938db4a33078b20ea6